src/fold.rs

```diff
@@ -126,19 +126,32 @@ impl Expr {
                 ExprType::Literal(UnsignedInt(sizeof))
             }
             ExprType::Negate(expr) => expr.const_fold()?.map_literal(
+                &location,
                 |token| match token {
-                    Int(i) => Int(-i),
-                    UnsignedInt(u) => UnsignedInt(0u64.wrapping_sub(u)),
-                    Char(c) => Char(0u8.wrapping_sub(c)),
-                    Float(f) => Float(-f),
-                    _ => token,
+                    Int(i) => {
+                        let (value, overflowed) = i.overflowing_neg();
```

Comment:
If you think it's worth it, you could abstract this into a function and reuse it for `mod`. NBD if not though.